anvil/src/eth/sign.rs

```diff
@@ -88,13 +88,11 @@ impl Signer for DevSigner {
         &self,
         request: TypedTransactionRequest,
         address: &Address,
-    ) -> Result<TypedTransaction, BlockchainError> {
+    ) -> Result<Signature, BlockchainError> {
         let signer = self.accounts.get(address).ok_or(BlockchainError::NoSignerAvailable)?;
         let ethers_tx: EthersTypedTransactionRequest = request.clone().into();
 
-        let signature = signer.sign_transaction_sync(&ethers_tx)?;
-
-        build_typed_transaction(request, signature)
```

Comment:
Moved this outside of `sign_transaction` and into it's [caller](https://github.com/foundry-rs/foundry/pull/5278/files#diff-bb6d14b47ca4f599f2dec954a44e9f3fbe244408940863b37f1f77dd861a70b7R368). Makes sense for `sign_transaction` to just return a signature?

Reply:
good call